skhep/math/vectors.py

```diff
@@ -684,37 +666,38 @@ def setptetaphie(self, pt, eta, phi, e):
         px, py, pz = pt*cos(phi), pt*sin(phi), pt*sinh(eta)
         self.setpxpypze(px,py,pz,e)
         
+    def tolist(self):
+        """Return the LorentzVector as a list."""
+        return list(self.__vector3d) + [self.__t]
+        
     def __setitem__(self, i, value):
         """Update/set the ith vector component (commencing at 0, of course)."""
         try:
-            self.__values[i] = value
+            if i == 3: self.__t = value
+            else: self.__vector3d[i] = value
         except IndexError:
             raise IndexError(
-                'Vector3D is of length {0} only!'.format(len(self)))
+                'LorentzVector is of length {0} only!'.format(len(self)))
 
     def __getitem__(self, i):
         """Get the ith vector component (commencing at 0, of course)."""
         try:
-            return self.__values[i]
+            return self.tolist()[i]
         except IndexError:
             raise IndexError( 'LorentzVector is of length {0} only!'.format(len(self)))
 
-    def tolist(self):
-        """Return the LorentzVector as a list."""
-        return list(self.__values)
-
     def __len__(self):
         """Length of the LorentzVector, i.e. the number of elements = 4."""
-        return len(self.__values)
+        return len(self.tolist())
         
     @property
     def p(self):
-        """Return the momentum, aka norm of the momentum 3D-vector."""
-        return self.smag
+        """Return the momentum, aka norm of the momentum Vector3D."""
```

Comment:
Cf. the comment above: the fact that the spatial component is a Vector3D is not the main thing. The message is indeed to return the norm of the spatial component, so the 3D-vector. Maybe "spatial component" is better.
Note that this would need to be changed in a few other places too.

Reply:
Noted!